include/vast/CodeGen/CodeGenAttrVisitor.hpp

```diff
@@ -65,5 +65,26 @@ namespace vast::cg {
         mlir_attr VisitWarnUnusedResultAttr(const clang::WarnUnusedResultAttr *attr) {
             return make< hl::WarnUnusedResultAttr >();
         }
+
+        mlir_attr VisitRestrictAttr(const clang::RestrictAttr *attr) {
+            return make< hl::MallocAttr >();
```

Comment:
is this typo? why not `RestrictAttr`?

Reply:
Nope, `RestrictAttr` is the name Clang uses for the `malloc` attribute… And it's even listed as such in their [docs](https://clang.llvm.org/docs/AttributeReference.html#malloc).
I named it malloc since that's how I discovered it in the source code I was using and the spelling of the attribute there is `malloc`. I can change it if you think it will be better.

Reply:
let's keep clangs naming for consistency